quic/s2n-quic-transport/src/space/datagram.rs

```diff
@@ -0,0 +1,88 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    endpoint,
+    stream::{AbstractStreamManager, StreamTrait as Stream},
+    transmission::WriteContext,
+};
+use s2n_quic_core::{
+    datagram::{Endpoint, Sender},
+    frame,
+};
+
+pub struct Manager<Config: endpoint::Config> {
+    sender: <<Config as endpoint::Config>::DatagramEndpoint as Endpoint>::Sender,
+    // TODO: Remove this warning once Receiver is implemented
+    #[allow(dead_code)]
+    receiver: <<Config as endpoint::Config>::DatagramEndpoint as Endpoint>::Receiver,
+}
+
+impl<Config: endpoint::Config> Manager<Config> {
+    pub fn new(
+        sender: <<Config as endpoint::Config>::DatagramEndpoint as Endpoint>::Sender,
+        receiver: <<Config as endpoint::Config>::DatagramEndpoint as Endpoint>::Receiver,
+    ) -> Self {
+        Self { sender, receiver }
+    }
+
+    // This function creates a packet that a user can interrogate to learn
+    // more about the packet space available for datagrams, and to write
+    // datagrams to the packet.
+    pub fn on_transmit<S: Stream, W: WriteContext>(
```

Comment:
Sry realized that my comment was on this function but I meant it more generally for this file. Specifically the `struct Manager`.

The on_transmit function is generally pretty clear imo.

```suggestion
    /// A callback that allows users to write datagrams directly to the packet.
    pub fn on_transmit<S: Stream, W: WriteContext>(
```